src/executables/sysrepo-plugind.c

```diff
@@ -323,13 +336,96 @@ load_plugins(struct srpd_plugin_s **plugins, int *plugin_count)
         /* finally store the plugin */
         (*plugins)[*plugin_count].handle = handle;
         (*plugins)[*plugin_count].private_data = NULL;
+
+        name_len = length_without_extension(ent->d_name);
+        if (name_len == 0) {
+            error_print(0, "Wrong filename \"%s\".", ent->d_name);
+            dlclose(handle);
+            rc = -1;
+            break;
+        }
+
+        new_str = strndup(ent->d_name, name_len);
```

Comment:
`new_str` is useless, no? You can assign directly to `plugin_name`, it will be set to `NULL` on error.